### ISSUE TITLE
[prefer-optional-chain] false positive for `T | boolean` types

### ISSUE THREAD

**Issue by the21st** [typescript-eslint/typescript-eslint#1893]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-optional-chain": ["error"]
  }
}
```

```TS
const x: string | false = foo()
if (x && x.includes('hello')) {
  console.log(x)
}
```

**Expected Result**
No linting issue is raised

**Actual Result**
Linter raises error

**Additional Info**
Autofix or manual fix to optional chaining results in:

```
Property 'includes' does not exist on type 'string | false'.
  Property 'includes' does not exist on type 'false'.ts(2339)
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.27.0` |
| `@typescript-eslint/parser`        | `2.27.0` |
| `TypeScript`                       | `3.7.4` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.16.0` |
| `npm`                              | `6.14.4` |

**Comment by sudalqueen** [comment]
May I try on this?😊

**Comment by bradzacher** [comment]
So the problem with this is that the only way to fix this is to introduce type information into the rule.
Right now it doesn't use it, so this will require a breaking change to fix properly.
Happy if you want to give it a go, but just as a heads up that merging the fix will have to wait for a major bump.

This is one of the reasons that we removed the autofixer from the rule - because without type information it can false-positive.

**Comment by sudalqueen** [comment]
@bradzacher Oh This was a bigger problem than I thought.
I think I'd better try it after I get more used to the typescript-eslint project.
Thanks for the comment!

**Comment by dasa** [comment]
The [appears](https://typescript-eslint.io/play/#ts=5.2.2&fileType=.ts&code=MYewdgzgLgBAHgLhtATgSzAcxgHxgMwEMAbCAUxgF4ZNiQAjEgFQAs0IA6fEEACgEoAUGnwxecGADJJ8DhmDEArgBMyEXgHIWZYnQ39%2BMAN6CYMUJBDEyHOpnFCAvoKA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WloHsalfkwCG8WmQAWo5uii9o-aJHBgAviHVA&tsconfig=&tokens=false) to be fixed.

**Comment by Josh-Cena** [comment]
Probably fixed by https://github.com/typescript-eslint/typescript-eslint/pull/6397
